frappe/core/doctype/docshare/docshare.py

```diff
@@ -40,6 +40,10 @@ def check_share_permission(self):
 			not frappe.has_permission(self.share_doctype, "share", self.get_doc())):
 
 			frappe.throw(_('You need to have "Share" permission'), frappe.PermissionError)
+			# if owner has the permission to write then only allow the self.user to write permission
```

Comment:
What is the user case for this? If I as a user have "write" permission, then I should be able to give "write" permission to others. Why should it be dependent on the "owner" of the document?